File: 백준/Silver/1931. 회의실 배정/회의실 배정.py
import sys


input = sys.stdin.readline
n = int(input())
meetings = [list(map(int, input().split())) for _ in range(n)]

meetings.sort(key=lambda x: (x[1], x[0]))

end_time = 0
count = 0

for start, end in meetings:
    if start >= end_time:
        end_time = end
        count += 1

print(count)

File: 백준/Silver/1931. 회의실 배정/회의실 배정.lua
local n = io.read("n")
local given = {}
for _ = 1, n do
    s, e = io.read("n", "n")
    table.insert(given, {s, e})
end

table.sort(given, function(a, b)
    if a[2] == b[2] then
        return a[1] < b[1]
    end
    return a[2] < b[2]
end)

local count = 1
local end_time = given[1][2]

for i = 2, n do
    local s, e = given[i][1], given[i][2]

    if s >= end_time then
        end_time = e
        count = count + 1
    end
end

print(count)
